Source/Artifacts/IArtifacts.cs

```diff
@@ -10,6 +12,11 @@ namespace Dolittle.SDK.Artifacts
     /// </summary>
     public interface IArtifacts
     {
+        /// <summary>
+        /// Gets the <see cref="BehaviorSubject{T}" /> of associated artifacts.
+        /// </summary>
+        BehaviorSubject<IDictionary<Type, Artifact>> Associations { get; }
```

Comment:
...Subject implementations should not be exposed to the outside - they are just useful implementations of both IObservable and IObserver with usefull behaviours. Exposing the ...Subject allows outside things to emit values out from the subject as well, which is usually not what you want.
```suggestion
        IObservable<IDictionary<Type, Artifact>> Associations { get; }
```

Reply:
I thought that exposing the IObserver part could be cook. But maybe that adds too much complexity